src/shared/codelens/csharpCodeLensProvider.ts

```diff
@@ -61,24 +83,80 @@ export async function initialize({
  * The command that is run when user clicks on Run Local or Debug Local CodeLens
  * @param commandName - Name of the VS Code Command currently running
  * @param lambdaLocalInvokeParams - Information about the Lambda Handler to invoke locally
+ * @param processInvoker - SAM CLI Process invoker
+ * @param taskInvoker - SAM CLI Task invoker
  */
 async function onLocalInvokeCommand({
+    configuration,
+    toolkitOutputChannel,
     commandName,
     lambdaLocalInvokeParams,
+    processInvoker,
+    taskInvoker,
+    telemetryService
 }: {
+    configuration: SettingsConfiguration
+    toolkitOutputChannel: vscode.OutputChannel,
     commandName: string,
     lambdaLocalInvokeParams: LambdaLocalInvokeParams,
+    processInvoker: SamCliProcessInvoker,
+    taskInvoker: SamCliTaskInvoker,
+    telemetryService: TelemetryService
 }): Promise<{ datum: Datum }> {
 
+    const channelLogger = getChannelLogger(toolkitOutputChannel)
     const runtime = await getRuntimeForLambda({
         handlerName: lambdaLocalInvokeParams.handlerName,
         templatePath: lambdaLocalInvokeParams.samTemplate.fsPath
     })
 
-    // TODO : Implement local run/debug in future backlog tasks
-    vscode.window.showInformationMessage(
-        `Local ${lambdaLocalInvokeParams.isDebug ? 'debug' : 'run'} support for csharp is currently not implemented.`
-    )
+    const baseBuildDir = await makeBuildDir()
```

Comment:
Add this to `ExtensionDisposableFiles` to ensure that it is cleaned up.

In the future, we should update all of our code lenses to clean the build dir after run/invoke, rather than waiting until the toolkit is shut down.

Reply:
Can do, also created an issue: https://github.com/aws/aws-toolkit-vscode/issues/489

Reply:
Looks like `makeBuildDir()` already does this. We can decide whether or not we still need the issue (theoretically, this won't affect new users, but the issue does cover log cleanup)

Reply:
Ah, perfect! Let's keep the issue, as we still want to consider cleaning up more aggressively.